components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -145,41 +149,8 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       if (shouldChangeBackgroundImage) {
         newState.backgroundImage = backgroundAPI.randomBackgroundImage()
       }
-      state = newState
-      break
-
-    case types.REMOVE_STACK_WIDGET:
-      const widget: NewTab.StackWidget = payload.widget
-      let { removedStackWidgets, widgetStackOrder } = state
-
-      if (!widgetStackOrder.length) {
-        break
-      }
-
-      if (!removedStackWidgets.includes(widget)) {
-        removedStackWidgets.push(widget)
-      }
-
-      state = {
-        ...state,
-        removedStackWidgets
-      }
-      break
-
-    case types.SET_FOREGROUND_STACK_WIDGET:
-      const frontWidget: NewTab.StackWidget = payload.widget
-      let newWidgetStackOrder = state.widgetStackOrder
-
-      newWidgetStackOrder = newWidgetStackOrder.filter((widget: NewTab.StackWidget) => {
-        return widget !== frontWidget
-      })
-
-      newWidgetStackOrder.push(frontWidget)
-
-      state = {
-        ...state,
-        widgetStackOrder: newWidgetStackOrder
-      }
+      // Handle updated widget prefs
+      state = handleWidgetPrefsChange(newState, state)
```

Comment:
Might be cleaner to just additionally handle `NEW_TAB_PREFERENCES_UPDATED` inside the `stackWidgetReducer` since the logic is isolated. No need to export the logic in to this reducer as far as I can tell.